### DISCUSSION TITLE
Common Approach to 3D Assets

### DISCUSSION THREAD

**Discussion by eldinor** [jelster/space-truckers#26] [General]
#### For each publishing GLTF asset:

-   File Size: Ideally less than 5MB. Note glTF geometry and texture compression extensions, such as glTF Universal Textures using the KTX container and geometry compression using Draco.
-   Draw calls: should be minimized by consolidating meshes, and using fewer materials.
-   Triangle Count: 100,000 triangles or less.
-   Texture Aspect Ratio: use power of 2 resolutions, square aspect ratio is not required.
-   Texture Size: Use  `1024*1024`  (1K) or  `2048*2048`  (2K) for BaseColor, ORM and Emissive maps. 2K is recommended for Normal maps which are more sensitive to reduced resolutions than even Albedo maps. Normals Maps are also severely sensitive to JPG artifacts - a 2K JPG giving the same quality as same as 1K PNG normal map.

Full guidelines at https://github.com/KhronosGroup/3DC-Asset-Creation/blob/main/asset-creation-guidelines/RealtimeAssetCreationGuidelines.md

**Comment by jelster** [comment]
My thought as well was to go with GLTF to allow separate texturing workflows, but for production builds is there better compression that could make glb a better choice?

**Reply by eldinor** [reply]
GLB files are more convenient to handle, in broad sense, since all assets are already inside them.
But GLTF allows to load textures async.
There are also cases where we can share the same cached textures between different meshes.
For texture compression there is KTX compression.
For geometry compression - Draco compression.

There could be almost no difference between GLB and GLTF total size, but if GLTF will let us achieve better performance I would choose GLTF.

**Reply by jelster** [reply]
Nice distinction on the ability to load gltf async and texture reuse
